compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -474,4 +487,58 @@ private void checkUniqueness(BLangConstant constant) {
             }
         }
     }
+
+    private void updateSymbolType(BLangConstant constant) {
+        if (constant.symbol.kind == SymbolKind.CONSTANT && constant.symbol.type.getKind() != TypeKind.FINITE &&
```

Comment:
Can you give an example for when the kind is not CONSTANT?

Reply:
My bad. It is a kind of unnecessary condition. Will remove. Thanks.